File: audience/src/main/java/br/com/concretesolutions/audience/core/concurrent/BaseHandler.java
package br.com.concretesolutions.audience.core.concurrent;

import android.os.Handler;
import android.os.Looper;
import android.os.Message;

import br.com.concretesolutions.audience.core.Director;
import br.com.concretesolutions.audience.core.MessageEvent;

public abstract class BaseHandler extends Handler {

    public BaseHandler(Looper looper) {
        super(looper);
    }

    @Override
    public void handleMessage(Message msg) {
        final MessageEvent event = (MessageEvent) msg.obj;
        Director.actorRegistry()
                .storageForRef(event.receiver)
                .executeEvent(event);
    }
}

File: audience-core/src/main/kotlin/io/github/victorolinasc/audience/core/concurrent/BaseHandler.kt
package io.github.victorolinasc.audience.core.concurrent

import android.os.Handler
import android.os.Looper
import android.os.Message

import io.github.victorolinasc.audience.core.Director
import io.github.victorolinasc.audience.core.MessageEvent

abstract class BaseHandler(looper: Looper) : Handler(looper) {

    override fun handleMessage(msg: Message) {
        val event = msg.obj as MessageEvent<*>
        Director.system.actorRegistry
                .storageForRef(event.receiver)
                .executeEvent(event)
    }
}
